ros/src/computing/planning/motion/packages/op_local_planner/nodes/op_behavior_selector/op_behavior_selector_core.cpp

```diff
@@ -0,0 +1,495 @@
+	sim_data.poses.push_back(p_pose);
+	sim_data.poses.push_back(p_box);
+	pub_SimuBoxPose.publish(sim_data);
+
+	//Send Trajectory Data to path following nodes
+	std_msgs::Int32 closest_waypoint;
+	PlannerHNS::RelativeInfo info;
+	PlannerHNS::PlanningHelpers::GetRelativeInfo(m_BehaviorGenerator.m_Path, m_BehaviorGenerator.state, info);
+	PlannerHNS::RosHelpers::ConvertFromLocalLaneToAutowareLane(m_BehaviorGenerator.m_Path, m_CurrentTrajectoryToSend, info.iBack);
+	//std::cout << "Path Size: " << m_BehaviorGenerator.m_Path.size() << ", Send Size: " << m_CurrentTrajectoryToSend << std::endl;
+
+	closest_waypoint.data = 1;
+	pub_ClosestIndex.publish(closest_waypoint);
+	pub_LocalBasePath.publish(m_CurrentTrajectoryToSend);
+	pub_LocalPath.publish(m_CurrentTrajectoryToSend);
+}
+
+void BehaviorGen::LogLocalPlanningInfo(double dt)
+{
+	timespec log_t;
+	UtilityHNS::UtilityH::GetTickCount(log_t);
+	std::ostringstream dataLine;
+	dataLine << UtilityHNS::UtilityH::GetLongTime(log_t) <<"," << dt << "," << m_CurrentBehavior.state << ","<< PlannerHNS::RosHelpers::GetBehaviorNameFromCode(m_CurrentBehavior.state) << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->m_pParams->rollOutNumber << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->bFullyBlock << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->iCentralTrajectory << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->iCurrSafeTrajectory << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->currentStopSignID << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->currentTrafficLightID << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->minStoppingDistance << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->distanceToNext << "," <<
+			m_BehaviorGenerator.m_pCurrentBehaviorState->GetCalcParams()->velocityOfNext << "," <<
+			m_CurrentBehavior.maxVelocity << "," <<
+			m_Twist_raw.twist.linear.x << "," <<
+			m_Twist_cmd.twist.linear.x << "," <<
+			m_Ctrl_cmd.linear_velocity << "," <<
+			m_VehicleStatus.speed << "," <<
+			m_VehicleStatus.steer << "," <<
+			m_BehaviorGenerator.state.pos.x << "," << m_BehaviorGenerator.state.pos.y << "," << m_BehaviorGenerator.state.pos.z << "," << UtilityHNS::UtilityH::SplitPositiveAngle(m_BehaviorGenerator.state.pos.a)+M_PI << ",";
+	m_LogData.push_back(dataLine.str());
+
+
+	if(bWayGlobalPath)
+	{
+		for(unsigned int i=0; i < m_GlobalPaths.size(); i++)
+		{
+			std::ostringstream str_out;
+			str_out << UtilityHNS::UtilityH::GetHomeDirectory();
+			str_out << UtilityHNS::DataRW::LoggingMainfolderName;
+			str_out << UtilityHNS::DataRW::PathLogFolderName;
+			str_out << "Global_Vel";
+			str_out << i;
+			str_out << "_";
+			PlannerHNS::PlanningHelpers::WritePathToFile(str_out.str(), m_GlobalPaths.at(i));
+		}
+		bWayGlobalPath = false;
+	}
+}
+
+void BehaviorGen::MainLoop()
+{
+	ros::Rate loop_rate(100);
```

Comment:
why 100?  could be a param?

Reply:
yes it should :) 
but it needs to be Autoware related system param.